editors/code/src/scopes_mapper.ts

```diff
@@ -0,0 +1,79 @@
+import * as vscode from 'vscode';
+import { TextMateRuleSettings } from './scopes';
+
+let mappings = new Map<string, string[]>();
+
+const defaultMapping = new Map<string, string[]>([
+    [
+        'comment',
```

Comment:
@matklad It looks like TM scopes are more expressive than RA tags. I suppose it wouldn't hurt to move towards the TM structure, even if not necessarily in this PR.

Reply:
I actually think RA scopes are better :) It would be cool with more granular scopes for Rust features alone. Especially for constraint bounds and lifetimes!